Assets/CohortUnityClient/Scripts/CHMessage.cs

```diff
@@ -1,29 +1,95 @@
 ﻿// Copyright Jacob Niedzwiecki, 2019
 // Released under the MIT License (see /LICENSE)
 
+using System;
 using System.Collections.Generic;
 
+
 namespace Cohort {
-  public class CHMessage {
-    public List<string> targetTags;
-    public MediaDomain mediaDomain;
-    public float cueNumber;
-    public CueAction cueAction;
-    public int id;
-    public string cueContent;
-
-    public static string FormattedMessage(CHMessage msg) {
-      string tagsForPrint = "";
-      msg.targetTags.ForEach(tag => tagsForPrint = tagsForPrint + tag + ", ");
-
-      return "    " + "cohort message: "
-        + "\n      tags:         " + tagsForPrint
-        + "\n      media domain: " + msg.mediaDomain
-        + "\n      cue #:        " + msg.cueNumber
-        + "\n      action:       " + msg.cueAction;
-    }
+    [Serializable]
+    public class CHMessage {
+
+        public List<string> targetTags;
+        public MediaDomain mediaDomain;
+        public double cueNumber;
+        public CueAction cueAction;
+        public int id;
+        public string cueContent;
+
+        public static string FormattedMessage(CHMessage msg) {
+          string tagsForPrint = "";
+          msg.targetTags.ForEach(tag => tagsForPrint = tagsForPrint + tag + ", ");
+
+          return "    " + "cohort message: "
+            + "\n      tags:         " + tagsForPrint
+            + "\n      media domain: " + msg.mediaDomain
+            + "\n      cue #:        " + msg.cueNumber
+            + "\n      action:       " + msg.cueAction;
+        }
+
+        public CHMessage FromSoundCue(CHSoundCue cue) {
+          Cue parsedCue;
+          parsedCue.mediaDomain = MediaDomain.sound;
+          parsedCue.cueNumber = cue.cueNumber;
+          parsedCue.cueAction = CueAction.play;
```

Comment:
Hmm, right, a CHSoundCue (video/image etc) doesn't have a cueAction set. This makes the conceptual / modelling gap I papered over much clearer... we should think about if there's clearer language we could adopt around 'the objects that live on the device that have a domain and a number and sometimes an asset' vs. 'the messages that get sent to devices that have a domain and a number and an action and tags and cause something to happen'...

Reply:
Yeah, cause right now you can't "build" a full cue in the inspector. It's more that you can set a piece of media up to receive a cue. So there's a bit of staging/faking going on here.